File: Concurs Inot C#/ValidatorProba.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Problema_3_CSharp.me.model;
using Problema_3_CSharp.me.exceptii;

namespace Problema_3_CSharp.me.validator
{
    public class ValidatorProba : IValidator<Proba>
    {
        public void Validate(Proba p)
        {
            String erori = "";
            if(p==null)
                throw new ValidationException("Obiectul nu poate fi null!\n");
            if (p.Distanta < 0)
                erori += "Distanta nu poate fi negativa!\n";
            if (!erori.Equals(""))
                throw new ValidationException(erori);
        }
    }
}

File: Concurs Inot Java/ValidatorProba.java
package me.validator;

import me.exceptii.ValidationException;
import me.model.Proba;

public class ValidatorProba implements Validator<Proba> {
    public void validate(Proba p) throws ValidationException{
        String erori="";
        if(p==null)
            throw new ValidationException("Obiectul nu exista!\n");
        if(p.getStil()==null)
            throw new ValidationException("Campurile nu pot fi null!\n");
        if(p.getDistanta()<0)
            erori+="Distanta nu poate fi negativa!\n";
        if(!erori.equals(""))
            throw new ValidationException(erori);
    }
}
